Sources/TiqrCoreObjC/Classes/IdentityService.m

```diff
@@ -71,6 +71,28 @@ - (void)deleteIdentityProvider:(IdentityProvider *)identityProvider {
     [self deleteObject:identityProvider];
 }
 
+- (void)deleteAllIdentitiesAndProviders: (NSError *)error {
+    // Delete the identities first
+    NSFetchRequest *identityRequest = [[NSFetchRequest alloc] initWithEntityName:@"Identity"];
+    NSBatchDeleteRequest *deleteIndentities = [[NSBatchDeleteRequest alloc] initWithFetchRequest:identityRequest];
+
+    [self.persistentStoreCoordinator executeRequest: deleteIndentities
+                                   withContext: self.managedObjectContext
+                                         error: &error];
+    if (error != nil) {
```

Comment:
executeRequest should return a boolean or a value. Only look at the error if you get back NO or nil. (It's kinda odd, but that's how Obj-C works.)